### DISCUSSION TITLE
Cannot Make Requests to Ghost Animator

### DISCUSSION THREAD

**Discussion by vlombardi22** [cmu-sei/GHOSTS-ANIMATOR#5] [Q&A] [answered]
I followed the tutorial for setup but cannot get the swagger page to work properly. Specifically, every time I try to post from it I get

"Not implemented: Ghosts.Animator.Models.InsiderThreat.InsiderThreatProfile"

Following that the request times out. Any advise?

**Comment by sei-dupdyke** [comment]
Hello, sorry for the issue - I just tried this and I see the insider threat error in the logs, but it still generated NPCs successfully.

How are you running Animator? In docker? 
Is your Mongo instance up and taking connections?
Are there any other errors you see before the insider threat one? (esp. at startup)

Thanks.

**Reply by vlombardi22** [reply]
I am running animator in docker just like the quick start section in the read me specifies. The mongo instance says it is waiting for connections. 

On startup I get an 

ghosts-animator-01  | warn: Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository[60]
ghosts-animator-01  |       Storing keys in a directory '/root/.aspnet/DataProtection-Keys' that may not be persisted outside of the container. Protected data will be unavailable when container is destroyed.


I get this right before the insider threat one
ghosts-animator-01  | warn: Microsoft.AspNetCore.HttpsPolicy.HttpsRedirectionMiddleware[3]
ghosts-animator-01  |       Failed to determine the https port for redirect.

Following the insider threat message I get a very long system time out message.

**Reply by sei-dupdyke** [reply]
Hmm, I can't replicate this. Are you starting the container stack via docker-compose?

**Reply by vlombardi22** [reply]
I have done it both with docker-compose and docker compose like it specifies in the tutorial. Similar problem every time.

**Reply by sei-dupdyke** [reply]
Hmm, I just deleted all images and tried it again with success. Once you run docker-compose up -d and browse to http://localhost:6001/swagger/index.html does that render as expected?

If it does, go to build POST
![image](https://user-images.githubusercontent.com/43444464/201424375-0b58cd2b-d073-488f-b6a5-cd0dc8644ef7.png)

and run this as it is. Then run the GET. Does it return created NPCs? Do you see any errors in the swagger page?

**Comment by sei-dupdyke** [comment, marked as answer]
I found the issue. Please add a hosts file addition on your host box mapping `host.docker.internal` to `127.0.0.1`. I will fix this in an upcoming release of the container. Sorry for the confusion!
